app/services/katello/pulp3/repository.rb

```diff
@@ -375,17 +405,68 @@ def create_distribution(path)
         distributions_api.create(distribution_data)
       end
 
-      def delete_distribution(href)
-        distributions_api.delete(href)
+      def self.delete_distribution(smart_proxy, href)
+        distributions_api(smart_proxy).delete(href)
       rescue api_exception_class => e
         raise e if e.code != 404
         nil
       end
 
+      def delete_distribution(href)
+        self.class.delete_distribution(smart_proxy, href)
+      end
+
       def lookup_distributions(args)
         distributions_api.list(args).results
       end
 
+      def self.distributions_list(smart_proxy, args = {})
+        fetch_from_list do |page_opts|
+          distributions_api(smart_proxy).list(page_opts.merge(args))
+        end
+      end
+
+      def distributions_list(args = {})
+        self.class.distributions_list(smart_proxy, args)
+      end
+
+      def self.pulp3_enabled_repo_types(smart_proxy, content_types = SETTINGS[:katello][:content_types].keys)
+        repository_types = content_types.collect do |content_type|
+          Katello::RepositoryTypeManager.find(content_type)
+        end
+
+        repository_types.select do |repository_type|
+          smart_proxy.pulp3_repository_type_support?(repository_type)
+        end
+      end
+
+      def self.delete_orphan_distributions(smart_proxy, repo_types = pulp3_enabled_repo_types(smart_proxy))
+        repo_types.each do |repo_type|
+          pulp3_class = repo_type.pulp3_service_class
+          orphan_distribution_hrefs = pulp3_class.orphan_distributions(smart_proxy)
+          orphan_distribution_hrefs.each do |distribution_href|
+            pulp3_class.delete_distribution(smart_proxy, distribution_href)
+          end
+        end
+      end
```

Comment:
This method needs to return a list of task objects of the work being done, so the dynflow action calling it can track it.    Otherwise i get:
```
ArgumentError

Missing the required parameter 'task_href' when calling TasksApi.read
```

when dynflow is trying to poll the tasks